airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -113,12 +120,19 @@ def hook(self) -> RedshiftDataHook:
 
     def execute(self, context: Context) -> GetStatementResultResponseTypeDef | str:
         """Execute a statement against Amazon Redshift."""
+        if self.cluster_identifier and self.workgroup_name:
+            raise AirflowException("Specify either 'cluster_identifier' or 'workgroup_name', not both.")
+
+        if not self.cluster_identifier and not self.workgroup_name:
+            raise AirflowException("Either 'cluster_identifier' or 'workgroup_name' must be specified.")
```

Comment:
You're already making these checks in the hook, no need to do it again here

Reply:
Thank you, addressed in 56cc6d8ea13031ca610250735c8b0071065b233f